src/main/resources/assets/computercraft/lua/rom/programs/edit.lua

```diff
@@ -135,39 +135,65 @@ local tKeywords = {
     ["while"] = true,
 }
 
-local function tryWrite( sLine, regex, colour )
-    local match = string.match( sLine, regex )
-    if match then
-        if type(colour) == "number" then
-            term.setTextColour( colour )
-        else
-            term.setTextColour( colour(match) )
-        end
-        term.write( match )
-        term.setTextColour( textColour )
-        return string.sub( sLine, string.len(match) + 1 )
-    end
-    return nil
-end
+local tHex = {
+    [ colors.white ] = "0",
+    [ colors.orange ] = "1",
+    [ colors.magenta ] = "2",
+    [ colors.lightBlue ] = "3",
+    [ colors.yellow ] = "4",
+    [ colors.lime ] = "5",
+    [ colors.pink ] = "6",
+    [ colors.gray ] = "7",
+    [ colors.lightGray ] = "8",
+    [ colors.cyan ] = "9",
+    [ colors.purple ] = "a",
+    [ colors.blue ] = "b",
+    [ colors.brown ] = "c",
+    [ colors.green ] = "d",
+    [ colors.red ] = "e",
+    [ colors.black ] = "f",
+}
+
+local tPattern = {
+    [ 1 ] = "^(%-%-%[%[.-%]%])()",
```

Comment:
Is there any reason for explicitly including the keys here, instead of using the normal list syntax?